### PR TITLE
Supports riak_kv changes for riak_kv#679

### PR DESCRIPTION
**Pull request by russelldb**
Some additions to vclock API to support creating fresh vclocks with a given timestamp.

A change to riak_core_util so that opening a non-existent file returns an error rather than crashing on a badmatch.

### CODE REVIEW

File: src/vclock.erl [hunk 1]
```diff
@@ -59,6 +60,11 @@ fresh() ->
 fresh(Node, Count) ->
     [{Node, {Count, timestamp()}}].
 
+-spec fresh(vclock_node(), counter(), timestamp()) ->
+                   vclock().
+fresh(Node, Count, Ts) ->
+    [{Node, {Count, Ts}}].
```
Comment (evanmcc) [inline src/vclock.erl:67]: do we want to do some structural validation here?  or will a bogus timestamp value immediately cause something else to crash?  what I am thinking of here is bad timestamps getting stored and then coming back later to explode in a manner hard to link to this change.  I am likely overthinking it.

Comment (russelldb) [inline src/vclock.erl:67]: No, not overthinking, I guess we should make sure it is a pos_integer().

Comment (seancribbs) [inline src/vclock.erl:67]: Guards would seem appropriate in this case.

Comment (russelldb) [inline src/vclock.erl:67]: Sure, I guess because it is not enforced at `increment/3` I didn't think to do it here. But I will

**Comment by slfritchie** [comment]
Ping?

**Comment by russelldb** [comment]
@slfritchie parked while we figure out what to do for riak_kv#679

**Comment by jrwest** [comment]
marked for milestone `2.1` since this all seems to be baking. someone more familiar @russelldb @slfritchie, can correct me, if need be.

**Comment by russelldb** [comment]
yeah, work restarted on this today. 2.1 seems right. Thanks @jrwest

**Comment by slfritchie** [comment]
+1

**Comment by russelldb** [comment]
Out of date. No longer needed, pertinent changes to riak_core_util merged already. Sorry all. Killing.

**Pull request closed without merging**